### PR TITLE
Automatically sign in users with stored credentials

### PR DESCRIPTION
**Pull request by rupertbates**
The aim of this PR is to remove a step from the membership 'become a supporter' flow for those users who have stored Guardian credentials.

Current implementation is to check if credentials exist and if they do then redirect to the sign in page where auto sign in will be carried out by existing code.

@markjamesbutler are you happy with this approach? It introduces an extra redirect but it has the benefit of being simple and keeping all the sign in code in one place. Alternatively I could refactor the sign in and register pages to extract the common functionality.

### CODE REVIEW

**Comment by markjamesbutler** [comment]
LGTM 👍

**Comment by markjamesbutler** [comment]
@rupertbates Would be good to make this work across the board, changes look good though. 👍

File: public/components/register-form/register-form.js [hunk 1]
```diff
@@ -143,3 +144,23 @@ export function init() {
 
   return form;
 }
+
+function checkForCredentials() {
+  //Only do this for the membership registration page
+  if (!window.location.search.includes("returnUrl=https://membership.theguardian.com"))
```
Comment (rupertbates) [inline public/components/register-form/register-form.js]: Yes it would work for all sites I was just being cautious, I'll make the change

**Comment by rupertbates** [comment]
@markjamesbutler I've made the changes you suggested, can you take a look please?

**Comment by markjamesbutler** [comment]
@rupertbates Looks good to me. 👍

**Pull request merged**